commit d8bf97a0a48901591f621bbed3de79bc4a6c8030
Author: Nathan Chancellor <natechancellor@gmail.com>
Date:   Mon Sep 24 10:33:44 2018 -0700

    cachefiles: Explicitly cast enumerated type in put_object
    
    [ Upstream commit b7e768b7e3522695ed36dcb48ecdcd344bd30a9b ]
    
    Clang warns when one enumerated type is implicitly converted to another.
    
    fs/cachefiles/namei.c:247:50: warning: implicit conversion from
    enumeration type 'enum cachefiles_obj_ref_trace' to different
    enumeration type 'enum fscache_obj_ref_trace' [-Wenum-conversion]
            cache->cache.ops->put_object(&xobject->fscache,
    cachefiles_obj_put_wait_retry);
    
    Silence this warning by explicitly casting to fscache_obj_ref_trace,
    which is also done in put_object.
    
    Reported-by: Nick Desaulniers <ndesaulniers@google.com>
    Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
    Signed-off-by: David Howells <dhowells@redhat.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/fs/cachefiles/namei.c b/fs/cachefiles/namei.c
index 95983c744164..5ab411d4bc59 100644
--- a/fs/cachefiles/namei.c
+++ b/fs/cachefiles/namei.c
@@ -244,11 +244,13 @@ static int cachefiles_mark_object_active(struct cachefiles_cache *cache,
 
 	ASSERT(!test_bit(CACHEFILES_OBJECT_ACTIVE, &xobject->flags));
 
-	cache->cache.ops->put_object(&xobject->fscache, cachefiles_obj_put_wait_retry);
+	cache->cache.ops->put_object(&xobject->fscache,
+		(enum fscache_obj_ref_trace)cachefiles_obj_put_wait_retry);
 	goto try_again;
 
 requeue:
-	cache->cache.ops->put_object(&xobject->fscache, cachefiles_obj_put_wait_timeo);
+	cache->cache.ops->put_object(&xobject->fscache,
+		(enum fscache_obj_ref_trace)cachefiles_obj_put_wait_timeo);
 	_leave(" = -ETIMEDOUT");
 	return -ETIMEDOUT;
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1117_d8bf97a0a48901591f621bbed3de79bc4a6c8030.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1117_d8bf97a0a48901591f621bbed3de79bc4a6c8030.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
